File: Add Code Here/C++/leetcode/longestPalindromicSubstring.cpp
string longestPalindrome(string s) {
     
        int n=s.length();
        int in=0;
        int m=1;
        
        for(int i=0;i<n;i++)
        {
            int l=i,r=i;
            
            while(l>=0&&r<n&&s[l]==s[r])
            {
                if(r-l+1>m)
                {
                    in=l;
                    m=r-l+1;
                }
                l--;r++;
            }
            
            l=i,r=i+1;
            
            while(l>=0&&r<n&&s[l]==s[r])
            {
                if(r-l+1>m)
                {
                    in=l;
                    m=r-l+1;
                }
                l--;r++;
            }
                
        }
        
        return s.substr(in,m);
        
    }

File: Add Code Here/JAVA/Java_Medium_Solutions/LongestPalindromicSubstring.java
// Problem Statement: Generate the longest possible substring from a given substring.

class Solution {
    public String longestPalindrome(String s) {
        int n = s.length();
        
        if(n <2){
            return s;
        }
        
        int start =0, end =0;
        for(int i=0; i< n; i++){
            char c =s.charAt(i);
            int j=i, k=i;
            while(j >=0 && s.charAt(j) == c){
                j--;
            }
            while(k<n && s.charAt(k) ==c){
                k++;
            }
            
            while(j >=0 && k <n){
                if(s.charAt(j) != s.charAt(k)){
                    break;
                }
                j--;
                k++;
            }
            j++;
            if(end -start < k-j){
                end =k;
                start =j;
            }
        }
        return s.substring(start, end);
    }
}
